lib/workers/repository/updates/branch-name.ts

```diff
@@ -42,16 +45,36 @@ export function generateBranchName(update: RenovateConfig): void {
       update.groupSlug = `patch-${update.groupSlug}`;
     }
     update.branchTopic = update.group.branchTopic || update.branchTopic;
-    update.branchName = template.compile(
-      update.group.branchName || update.branchName,
-      update
-    );
+    update.branchName = update.group.branchName || update.branchName;
+  }
+
+  if (update.hashedBranchLength) {
+    let hashLength = update.hashedBranchLength - update.branchPrefix.length;
+    if (hashLength <= MIN_HASH_LENGTH) {
+      logger.warn(
+        `\`hashedBranchLength\` must allow for at least 6 characters hashing in addition to \`branchPrefix\`. Using ${MIN_HASH_LENGTH} character hash instead.`
+      );
+      hashLength = MIN_HASH_LENGTH;
+    }
+
+    const additionalBranchPrefix = update.additionalBranchPrefix
+      ? template.compile(String(update.additionalBranchPrefix), update)
+      : '';
+
+    const branchTopic = update.branchTopic
+      ? template.compile(String(update.branchTopic), update)
+      : '';
+
+    const hash = hasha(additionalBranchPrefix + branchTopic);
```

Comment:
Don't you need to compile again just in case?

Reply:
done ✅